crypto/err/err.c

```diff
@@ -839,17 +721,7 @@ const char *ERR_reason_error_string(unsigned long e)
 
 void ERR_remove_thread_state(const CRYPTO_THREADID *id)
 {
-    ERR_STATE tmp;
-
-    if (id)
-        CRYPTO_THREADID_cpy(&tmp.tid, id);
-    else
-        CRYPTO_THREADID_current(&tmp.tid);
-    /*
-     * thread_del_item automatically destroys the LHASH if the number of
-     * items reaches zero.
-     */
-    int_thread_del_item(&tmp);
+    ERR_clear_error();
```

Comment:
Right. Should be ERR_get_state+ERR_STATE_free. I should also call CRYPTO_THREAD_set_local(..., NULL) so that the next call to ERR_get_state will initialize a new state as well.